pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -886,6 +889,16 @@ func proposeBatch(
 		//      slice of entries. See etcd-io/raft#57.
 		maybeDeductFlowTokens(ctx, p.flowControlHandle(ctx), handles, ents)
 	}
+
+	// If the context on the proposal is set up for tracing, register the
+	// proposal with rafttrace. This will add the trace to updates.
+	for i, e := range ents {
+		if s := tracing.SpanFromContext(props[i].Context()); s != nil && s.RecordingType() == tracingpb.RecordingVerbose {
```

Comment:
I don't know much about `RecordingType()`. Could you please comment why we only trace with `RecordingVerbose`?

Reply:
Added. I'm not 100% clear on it either and when we have the structured recordings, but I think it requires a different type of logging. I'll look into this a bit more, but for now I at least added a comment.